src/main/java/com/thealgorithms/searches/BinarySearch2dArray.java

```diff
@@ -73,13 +97,20 @@ static int[] binarySearch(
         int colEnd
     ) {
         while (colStart <= colEnd) {
+            branchesReached.add("8.1");
             int midIndex = colStart + (colEnd - colStart) / 2;
 
-            if (arr[row][midIndex] == target) return new int[] {
+            if (arr[row][midIndex] == target) {
+                branchesReached.add("8.1.1");
+                return new int[] {
                 row,
-                midIndex,
-            }; else if (arr[row][midIndex] < target) colStart =
-                midIndex + 1; else colEnd = midIndex - 1;
+                midIndex, }; 
+            } else if (arr[row][midIndex] < target) { 
+                branchesReached.add("8.1.2");
+                colStart = midIndex + 1; 
+            } else {
+                colEnd = midIndex - 1;
+            }
         }
```

Comment:
Coverage measurement should only concern the BinarySearch2dArray::BinarySearch method, and not BinarySearch2dArray::binarySearch (notice the initial capital B).

Reply:
@einbergisak fixed